pkg/datasource.go

```diff
@@ -0,0 +1,259 @@
+package main
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	simplejson "github.com/bitly/go-simplejson"
+	"github.com/grafana/grafana-plugin-model/go/datasource"
+	hclog "github.com/hashicorp/go-hclog"
+	plugin "github.com/hashicorp/go-plugin"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+type BQDatasource struct {
+	plugin.NetRPCUnsupportedPlugin
+	logger hclog.Logger
+}
+
+func (t *BQDatasource) Query(ctx context.Context, tsdbReq *datasource.DatasourceRequest) (*datasource.DatasourceResponse, error) {
+	t.logger.Debug("Query", "datasource", tsdbReq.Datasource.Name, "TimeRange", tsdbReq.TimeRange)
```

Comment:
Where can we see the debug messages after code is deployed?

Reply:
The logs are at grafana.log file